src/btree/bt_delete.c

```diff
@@ -322,25 +343,43 @@ __wt_delete_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref)
 	 * In the first case, we have a page reference structure, in the second
 	 * second, we don't.
 	 *
-	 * Allocate the per-reference update array; in the case of instantiating
-	 * a page, deleted by a running transaction that might eventually abort,
-	 * we need a list of the update structures so we can do that abort.  The
-	 * hard case is if a page splits: the update structures might be moved
-	 * to different pages, and we still have to find them all for an abort.
+	 * In the first case, check to see if the transaction has been resolved,
+	 * maybe we don't have to do that much work.
 	 */
+	page_del = ref->page_del;
+	if (page_del != NULL &&
+	    __wt_txn_visible_all(session,
+	    page_del->txnid, WT_TIMESTAMP_NULL(&page_del->timestamp)))
+		__wt_free(session, ref->page_del);
+
```

Comment:
@michaelcahill, would you please confirm this visibility test change? The idea is that if we're about to instantiate a deleted page, and the transaction that did the delete is now globally visible, we don't need to worry about the transaction aborting and we can skip a bunch of work.